### DISCUSSION TITLE
Suggestion: Add-MtTestResultDetail new parameter

### DISCUSSION THREAD

**Discussion by magnusjak** [maester365/maester#477] [Ideas]
I have been using the Add-MtTestResultDetail to create Markdown results as you have in most of your tests (using the -Result parameter) and this works great.

However, I find myself wanting to have another set of results that are more easily read programmatically. For example change the parameters to include -ResultMarkdown and -ResultJson (or just -Result). This way we can create specific Markdown results to be shown in the HTML report, AND we can have json, or other types of data-results saved in the testresult.json file for programmatic use if needed.

**Comment by merill** [comment]
This is a good idea. I'm moving this to Discussion to add to the backlog.

Can you share some examples of how you would like it to be split up? Right now the json export is mainly the output but we can split it out as needed.

**Reply by magnusjak** [reply]
Something like this in the .json output file perhaps?
```
{
    "Tests": [
        {
        "Name": "<testName>,
          "ResultDetail": {
              "TestResultMarkdown": "<markdown>",
              "TestResultJson": "<JsonObject>",
              "TestDescription": "<description>",
              "SkippedReason": null,
              "TestSkipped": ""
          }
        }
    ]
}
```

So the TestResultMarkdown might have a Markdown formatted value that shows a table with users that have either failed or passed a test. Then the TestResultJson would have the user-objects as pure JSON with perhaps more information than the Markdown table has room for.

Or, the only thing needed in the 'TestResult' in the JSON file would be the JSON-formatted output, while the Markdown formatted output is placed in the .html and .md output files.
